lib/simple_router.rb

```diff
@@ -1,7 +1,8 @@
 module SimpleRouter
   class << self
-    attr_accessor :engine
-    def engine() @engine || Engines::SimpleEngine end
+    attr_accessor :engine_reference
+    def engine() self.engine_reference || Engines::SimpleEngine end
+    def engine=(value) self.engine_reference=value end
```

Comment:
The reason you need `self.engine_reference=` and not just `engine_reference=` is that the interpreter will see the latter as an assignment to a local variable and issue an "set but not used" warning ...